### PR TITLE
chore(ci): disable playwright test detection script on merge queue

### PR DESCRIPTION
**Pull request by fpbrault**
This PR ensures that all playwright tests run for the merge queue instead of just those related to the files that where changed.

https://coveord.atlassian.net/browse/KIT-3702

### CODE REVIEW

File: .github/workflows/prbot.yml [hunk 1]
```diff
@@ -104,7 +104,13 @@ jobs:
...
       - name: Identify E2E Test Files to run
         id: determine-tests
-        run: node ./scripts/ci/determine-tests.mjs testsToRun shardIndex shardTotal
+        run: |
+
+          if [ "${{ github.event_name }}" == "merge_group" ]; then
+            node ./scripts/ci/determine-tests.mjs testsToRun shardIndex shardTotal true
+          else
+            node ./scripts/ci/determine-tests.mjs testsToRun shardIndex shardTotal
+          fi
```
Comment (alexprudhomme) [inline .github/workflows/prbot.yml:113]: I think this putting `false` would be clearer with the combination of `const runAllTests = process.argv[5]`
````suggestion
        run: |

          if [ "${{ github.event_name }}" == "merge_group" ]; then
            node ./scripts/ci/determine-tests.mjs testsToRun shardIndex shardTotal true
          else
            node ./scripts/ci/determine-tests.mjs testsToRun shardIndex shardTotal false
          fi
```

File: scripts/ci/determine-tests.mjs [hunk 2]
```diff
@@ -176,13 +176,16 @@ const changedFiles = getChangedFiles(base, head).split(EOL);
...
 const projectRoot = process.env.projectRoot;
 const atomicSourceComponents = join('packages', 'atomic', 'src', 'components');
 
 try {
   const testFiles = findAllTestFiles(atomicSourceComponents);
   const testDependencies = createTestFileMappings(testFiles, projectRoot);
-  const testsToRun = determineTestFilesToRun(changedFiles, testDependencies);
+  const testsToRun = runAllTests
+    ? testFiles.join(' ')
+    : determineTestFilesToRun(changedFiles, testDependencies);
```
Comment (alexprudhomme) [inline scripts/ci/determine-tests.mjs:188]: This will only work for the basic use case where tests need to be determined. The NoRelevantChangesError error below won't run all tests all the time in the merge queue.

File: scripts/ci/determine-tests.mjs [hunk 3]
```diff
@@ -176,14 +176,17 @@ const changedFiles = getChangedFiles(base, head).split(EOL);
...
   const testFiles = findAllTestFiles(atomicSourceComponents);
   const testDependencies = createTestFileMappings(testFiles, projectRoot);
-  const testsToRun = determineTestFilesToRun(changedFiles, testDependencies);
-  if (testsToRun === '') {
+  const testsToRun = runAllTests
+    ? testFiles.join(' ')
+    : determineTestFilesToRun(changedFiles, testDependencies);
+  if (testsToRun === '' && !runAllTests) {
     throw new NoRelevantChangesError();
   }
```
Comment (alexprudhomme) [inline scripts/ci/determine-tests.mjs:191]: This is hard to parse imo, I would put it this way very clearly and exit with an error if runAllTests is true before running all other functions. So at the very beginning of the try I would if check runAllTests and immediately throw. You can then catch it with the other 'run all tests' errors.

```
class RunAllTestsInMergeQueueError extends Error {
  constructor() {
    super(
      `In a merge queue. Running all tests.`
    );
    this.name = 'RunAllTestsInMergeQueueError';
  }
}
try {
if(runAllTests){
throw new RunAllTestsInMergeQueue();
}

  const testFiles = findAllTestFiles(atomicSourceComponents);
  const testDependencies = createTestFileMappings(testFiles, projectRoot);
  const testsToRun = determineTestFilesToRun(changedFiles, testDependencies);
  if (testsToRun === '') {
    throw new NoRelevantChangesError();
  }
  ...
  
  catch (error) {
...

  if (
    error instanceof DependentPackageChangeError ||
    error instanceof PackageJsonChangeError ||
    error instanceof RunAllTestsInMergeQueueError
  ) {
```

Comment (fpbrault) [reply scripts/ci/determine-tests.mjs:191]: Good point, I'll improve that

File: scripts/ci/determine-tests.mjs [hunk 4]
```diff
@@ -202,11 +215,10 @@ try {
...
     setOutput(outputNameShardIndex, [0]);
     setOutput(outputNameShardTotal, [0]);
-  }
-
-  if (
+  } else if (
     error instanceof DependentPackageChangeError ||
-    error instanceof PackageJsonChangeError
+    error instanceof PackageJsonChangeError ||
+    error instanceof RunAllTestsInMergeQueueError
```
Comment (louis-bompart) [inline scripts/ci/determine-tests.mjs:221]: Let's simplify :)
Add an abstract class in the proto chain that all those errors extends, instead of them extending.
Then, do an instanceof on this abstract class instead of having 3 instanceof.

**Review by louis-bompart — APPROVED** [review]
LGTM, nitpik to DRY

File: scripts/ci/determine-tests.mjs (hunk 4 above)
Comment (alexprudhomme) [reply scripts/ci/determine-tests.mjs:221]: Personally, I like it better like it is now. It is way more explicit. Why add more abstraction and complexity in a simple script ?

**Review by alexprudhomme — APPROVED** [review]

File: scripts/ci/determine-tests.mjs (hunk 4 above)
Comment (alexprudhomme) [reply scripts/ci/determine-tests.mjs:221]: The less lines the better 🤷
